### PR TITLE
Add Python3 support

### PR DESCRIPTION
**Pull request by vladislavkoz**
Updated Python version to >3

Added support for the following OS:
- CentOS8
- Ubuntu20.04(Focal Fossa)
- Ubuntu22.04(Jammy Jellyfish)
- Debian10(Buster)
- Debian11(BullSeye)
- Debian12(Bookworm)

Removed support for the following OS: 
- OS(glibc<2.17): 
- CentOS6
- Ubuntu12(Precise Pangolin)
- Ubuntu14(Trusty Tahr)
- Debian7(Wheezy)
- Debian8 (Jessie)
- Debian9 (Stretch)

Updated Dockerfile to use CentOS7 + Python3
Updated Vagrant testing script

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Editor-based HTTP Client requests
+/httpRequests/
+# Datasource local storage ignored files
+/dataSources/
+/dataSources.local.xml
```
Comment (NazarPo) [inline .idea/.gitignore:8]: Although I can see you added `.idea` to `.gitignore`, its content was added to this commit. Let's reset and `push -f`

File: sysvinit/root/etc/init.d/hg-agent [hunk 2]
```diff
@@ -1,68 +0,0 @@
-#! /bin/bash
```
Comment (NazarPo) [inline sysvinit/root/etc/init.d/hg-agent:1]: May I ask you to explain the reason for removing this entire file?

File: Makefile [hunk 3]
```diff
@@ -1,26 +1,28 @@
 NAME=hg-agent
-VERSION=1.18
+VERSION=2.0
 ARCH=amd64
 
 docker:
-	docker build -t hostedgraphite/hg-agent-build .
+	docker build -t hostedgraphite/hg-agent-build-os7-py3 .
```
Comment (NazarPo) [inline Makefile:6]: May I ask you to share with me some details on this `os7-py3` suffix? Does it have anything to do with Centos 7 OS? How did we get this image?

File: Dockerfile [hunk 4]
```diff
@@ -1,31 +1,35 @@
 # A build environment for the Hosted Graphite agent.
 # Because we use PyInstaller, we need to build against the oldest glibc we
 # intend to support: that's 2.12, available in CentOS 6.
```
Comment (NazarPo) [inline Dockerfile:3]: This comment must be updated or removed, don't you agree?

**Review by NazarPo — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 4 above)
Comment (vladislavkoz) [reply Dockerfile]: Updated
image

File: Makefile (hunk 3 above)
Comment (vladislavkoz) [reply Makefile:6]: This image was created on my local machine and pushed to the DockerHub.
image

Comment (vladislavkoz) [reply Makefile:6]: It uses Dockerfile from hg-agent folder.

File: sysvinit/root/etc/init.d/hg-agent (hunk 2 above)
Comment (vladislavkoz) [reply sysvinit/root/etc/init.d/hg-agent:1]: Removed support for:
CentOS6, Ubuntu12(Precise Pangolin), Debian7(Wheezy)

sysvinit was used on some old OS versions. We do not need it more.

Comment (NazarPo) [reply sysvinit/root/etc/init.d/hg-agent:1]: Oh, I see. Okay. Thanks

File: Makefile (hunk 3 above)
Comment (NazarPo) [reply Makefile:6]: Brilliant! Thanks for the explanation

**Comment by NazarPo** [comment]
The CI step `sudo apt-get update`:

image

Should we add a tech debt to Asana? This is similar to [this](https://app.asana.com/0/1201106364924220/1201863952022509) issue, but I assume this is something different and needs a separate ticket

**Comment by NazarPo** [comment]
The CI step `make build`:

image

Let's take a peek at this issue. This shouldn't take a lot of time, I guess

**Comment by vladislavkoz** [comment]
> The CI step `sudo apt-get update`:
> 
> image
> 
> Should we add a tech debt to Asana? This is similar to [this](https://app.asana.com/0/1201106364924220/1201863952022509) issue, but I assume this is something different and needs a separate ticket

Task added. https://app.asana.com/0/0/1201883327069631/f

**Review by NazarPo — APPROVED** [review]

**Pull request merged**